src/modules/ce/components/ClientEligibility.tsx

```diff
@@ -0,0 +1,134 @@
+import { Stack, Typography } from '@mui/material';
+import { useCallback, useContext, useState } from 'react';
+import LoadingButton from '@/components/elements/LoadingButton';
+import AssessmentContext from '@/modules/assessments/components/AssessmentContext';
+import ApolloErrorAlert from '@/modules/errors/components/ApolloErrorAlert';
+import ErrorAlert from '@/modules/errors/components/ErrorAlert';
+import { emptyErrorState, ErrorState, hasErrors } from '@/modules/errors/util';
+import { FormDefinitionHandlers } from '@/modules/form/hooks/useFormDefinitionHandlers';
+import { DynamicInputCommonProps } from '@/modules/form/types';
+import { localResolvePickList } from '@/modules/form/util/formUtil';
+import {
+  CalculateClientEligibilityMutation,
+  PickListOption,
+  useCalculateClientEligibilityMutation,
+} from '@/types/gqlTypes';
+
+const projectTypePickList = localResolvePickList('ProjectType');
+
+interface ClientEligibilityProps extends DynamicInputCommonProps {
+  value?: PickListOption[] | null;
+  onChange?: (value: PickListOption[] | null) => void;
+  handlers?: FormDefinitionHandlers;
+}
+
+// Shows a button to calculate client eligibility based on current form values, and displays the result.
+const ClientEligibility = ({
```

Comment:
I played around with a refactor to reduce shared code with `AltAhaScore` by creating a new generic `MutationButtonField`.

I didn't really _love_ the result -- just based on vibes, it seems less graceful and maintainable to me -- so I reverted it. That work is here if you are curious to check it out: https://github.com/greenriver/hmis-frontend/commit/e95b45e4b8e3fe4100233a7f51ea5ddbc6550c25 

I do still think we could introduce a pattern like this if we expect to start adding more and more form components that fit this mold, so let's discuss.